presto-main/src/main/java/com/facebook/presto/operator/HashSemiJoinOperator.java

```diff
@@ -173,8 +195,20 @@ public void addInput(Page page)
 
         // update hashing strategy to use probe cursor
         for (int position = 0; position < page.getPositionCount(); position++) {
-            if (probeJoinPage.getBlock(0).isNull(position)) {
-                blockBuilder.appendNull();
+            boolean isIndeterminate;
+            try {
+                isIndeterminate = probeJoinPage.getBlock(0).isNull(position) ||
+                        (boolean) probeIndeterminateFunction.invoke(readNativeValue(types.get(probeJoinChannel), probeJoinPage.getBlock(0), position), false);
+            }
+            catch (Throwable t) {
+                Throwables.propagateIfInstanceOf(t, Error.class);
+                Throwables.propagateIfInstanceOf(t, PrestoException.class);
+
+                throw new PrestoException(GENERIC_INTERNAL_ERROR, t);
+            }
+            if (isIndeterminate) {
+                // forbid the indeterminate values in probe side
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "indeterminate values are not allowed in probe side");
```

Comment:
This check is obsolete now, as `master` has b1c7711170ee3e032ed076340ec7e76d40c4584c commit doing quite the same (and in the same place)

> indeterminate values are not allowed in probe side

That's true. Until #7792